cfme/tests/containers/test_projects_dashboard.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+
+from cfme.containers.provider import ContainersProvider
+from cfme.containers.container import Container
+from cfme.containers.image import Image
+from cfme.containers.service import Service
+
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.version import current_version
+
+
+pytestmark = [
+    pytest.mark.uncollectif(lambda provider: current_version() < "5.9"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1),
+    pytest.mark.provider([ContainersProvider], scope='function')
+]
+
+TESTED_OBJECTS = [Container, Image, Service]
+PROJECT_NAME = 'test-project-dashboard'  # Predefine project for this test
+
+
+def get_api_object_counts(appliance, project_name, provider):
+    """ Fetches amount of Containers/Services/Images from the API per selected project name"""
+    assert isinstance(provider, ContainersProvider)
+    return {
+        Container: len(provider.mgmt.list_container(project_name=project_name)),
```

Comment:
I think if this return is going to be keyed on the classes, those class keys should be used in the test instead of TESTED_OBJECTS list.

```python
for cls in api_values.keys():
    <check UI value of cls.__name__ / cls.PLURAL etc>

Reply:
Basically the idea is to link the data sets somehow - right now this method generates a dictionary keyed on classes that are hard coded here.  If TESTED_OBJECTS is changed, but a new class isn't added here to match, then the test will have a mismatch it what its trying to compare.  The test should just use the class keys that are in the dictionary returned by this method, not some other arbitrary list of classes.

Reply:
Remove TESTED_OBJECTS, use api_values.keys() instead.